src/fn_call.rs

```diff
@@ -114,6 +114,8 @@ impl<'a, 'mir, 'tcx: 'mir + 'a> EvalContextExt<'tcx, 'mir> for super::MiriEvalCo
             None => self.tcx.item_name(def_id).as_str(),
         };
 
+        let tcx = &{self.tcx.tcx};
```

Comment:
???

Reply:
We cannot use `self` in the arguments of `&mut self` methods. But we do need a `tcx` in the arguments. The argument needs to be a `&impl HasDataLayout`, so I need a `&`. The `{}` around it is to make sure we're not borrowing `self` here.